v0.0.1/tasks-schema.json

```diff
@@ -0,0 +1,957 @@
+                      "examples": ["inc hosp", "peak week hosp"],
+                      "type": "string",
+                      "maxLength": 20
+                    },
+                    "target_name": {
+                      "description": "A longer human readable target description that could be used, for example, as a visualisation axis label.",
+                      "examples": [
+                        "Weekly incident influenza hospitalizations",
+                        "Peak week for incident influenza hospitalizations"
+                      ],
+                      "type": "string",
+                      "maxLength": 100
+                    },
+                    "target_units": {
+                      "description": "Unit of observation of the target.",
+                      "examples": [
+                        "rate per 100,000 population",
+                        "count",
+                        "date"
+                      ],
+                      "type": "string",
+                      "maxLength": 100
+                    },
+                    "target_keys": {
+                      "description": "Object that either is null or has 1 or more properties, the names of which match task_id variable(s) named within the same model_tasks object. Each property should have one specified value. When combined, the value(s) of these keys define the acceptable target value option",
+                      "oneOf": [{
+                        "examples": ["inc hosp", "peak week hosp"],
+                        "type": "object",
+                        "properties": {
+                          "target": {
+                            "description": "single column target ID descriptor",
+                            "type": "string"
+                          }
+                        }
+                      }, {
+                        "examples": [{
+                            "target_variable": "hosp",
+                            "target_outcome": "inc"
+                          },
+                          {
+                            "target_variable": "case",
+                            "target_outcome": "peak week"
+                          }
+                        ],
+                        "type": "object",
+                        "properties": {
+                          "target_variable": {
+                            "description": "Target ID column name containing the target variable.",
+                            "type": "string"
+                          },
+                          "target_measure": {
+                            "description": "Target ID column name containing the target outcome.",
+                            "type": "string"
+                          }
+                        }
+
+                      }, {
+                        "examples": "null",
+                        "type": "null"
+                      }]
```

Comment:
If I understand, we're being quite specific here that the column(s) identifying the target **must** be exactly one of the following options:

1. `null` (i.e., no column that identifies the target)
2. `target` (i.e., a column named "target" that identifies the target)
3. `target_variable` and `target_measure` (two columns with those names that in combination identify the target)

But I was thinking that the plan was to allow the names of the columns that identify the target to be set arbitrarily by the hub?

Reply:
OK, will make more flexible 👍

Reply:
In the end, because we want to make `target_keys` flexible and validation will be needed in R so handle in R instead.

Reply:
Resolved in [57205d6](https://github.com/Infectious-Disease-Modeling-Hubs/schemas/pull/28/commits/57205d6435f79a1644af7c76ba6abb24004be34b)